File: Data-Structures/Stacks/Stack/Stack.py
import msvcrt
import os

stack=[]
top=0
size=5

def main():
    choice=0
    while choice!=4:
        os.system('cls')
        print "\n 1. Push  "
        print " 2. Pop     "
        print " 3. Display "
        print " 4. Exit    "
        choice=int(input("\n Enter>>"))
        if choice==1:
            push()
        elif choice==2:
            pop()
        elif choice==3:
            display()

def push():
    global stack,top,size
    if top<size:
        top=top+1
        stack.append(input("\n Push value : "))
    else:
        print "\n Stack is overflowing!"
        char=msvcrt.getch()

def pop():
    global stack,top
    if top>0:
        print "\n Popped value :",stack.pop()
        top=top-1;
    else:
        print "\n Stack is underflowing!"
    char=msvcrt.getch()

def display():
    global stack,top
    if top>0:
        print "\n The stack is : \n"
        for i in reversed(stack):
            print "   ",i,"\n"
    else:
        print "\n Stack is underflowing!"
    char=msvcrt.getch()

main()

File: Data-Structures/Stacks/Stack/Stack.c
#include<stdio.h>
#include<conio.h>

int stack[10],top=0,size=5;

void push();
void pop();
void display();

void main()
{
    int choice;
    do
    {
        system("cls");
        printf("\n    MENU :- \n");
        printf("\n 1. Push \n");
        printf("\n 2. Pop \n");
        printf("\n 3. Display \n");
        printf("\n 4. Exit \n");
        printf("\n    Enter your choice : ");
        scanf("%d",&choice);
        switch(choice)
        {
            case 1 : push();    break;
            case 2 : pop();        break;
            case 3 : display();    break;
        }
    }while(choice!=4);
}

void push()
{
    if(top<size)
    {
        top=top+1;
        printf("\n    Push value : ");
        scanf("%d",&stack[top]);
    }
    else
    {
        printf("\n    Stack is overflowing!");
        getch();
    }
}

void pop()
{
    if(top>0)
    {
        printf("\n    Popped value : %d",stack[top]);
        top=top-1;
    }
    else printf("\n    Stack is underflowing!");
    getch();
}

void display()
{
    int i;
    if(top>0)
    {
        printf("\n    The stack is : ");
        for(i=top;i>0;i--)
            printf("\n\n    %d",stack[i]);
    }
    else printf("\n    Stack is underflowing!");
    getch();
}
